packages/rum-core/src/performance-monitoring/navigation/capture-navigation.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import {
+  PERF,
+  isPerfTimelineSupported,
+  isRedirectInfoAvailable
+} from '../../common/utils'
+import { PAGE_LOAD, RESOURCE, MEASURE } from '../../common/constants'
+import { state } from '../../state'
+import { createNavigationTimingSpans } from './navigation-timing-spans'
+import { createUserTimingSpans } from './user-timing-spans'
+import { createResourceTimingSpans } from './resource-timing-spans'
+import { getPageLoadMarks } from './marks'
+
+function captureNavigation(transaction) {
+  /**
+   * Do not capture timing related information when the
+   * flag is set to false, By default both page-load and route-change
+   * transactions set this flag to true
+   */
+  if (!transaction.captureTimings) {
+    return
+  }
+
+  /**
+   * Both start and end threshold decides if a span must be
+   * captured as part of the transaction
+   */
+  const trEnd = transaction._end
+  /**
+   * Page load is considered as hard navigation and we account
+   * for few extra spans than soft navigations which
+   * happens on single page applications
+   */
+  if (transaction.type === PAGE_LOAD) {
+    /**
+     * Adjust custom marks properly to fit in the transaction timeframe
+     */
+    if (transaction.marks && transaction.marks.custom) {
+      const customMarks = transaction.marks.custom
+      Object.keys(customMarks).forEach(key => {
+        customMarks[key] += transaction._start
+      })
+    }
+    /**
+     * must be zero otherwise the calculated relative _start time would be wrong
+     */
+    const trStart = 0
+    transaction._start = trStart
+
+    const timings = PERF.timing
+    const baseTime = isRedirectInfoAvailable(timings)
```

Comment:
also, this change makes possible that spans take the redirect duration into consideration